src/account/AccountManager.ts

```diff
@@ -0,0 +1,241 @@
+                error instanceof Error ? error.message : 'UNKNOWN'
+            );
+            throw error;
+        }
+    }
+
+    /**
+     * Updates the registration code status with exponential backoff retry logic
+     */
+    public static async updateRegCodeStatus(authToken: string): Promise<void> {
+        console.log(`${AccountManager.LOG_TAG}: In RegCode update`);
+        
+        const regCode = await Storage.getRegCode();
+        if (!regCode) {
+            throw new Error('RegCode not found');
+        }
+
+        const url = `${AccountManager.BASE_URL}/accountregcode/${regCode}/status`;
+        console.log(`${AccountManager.LOG_TAG}: fetch url = ${url}`);
+
+        const body = {
+            deviceId: await getDeviceId()
+        };
+
+        const headers = {
+            'Authorization': authToken,
+            'Content-Type': 'application/json'
+        };
+
+        try {
+            await AccountManager.executeWithRetry(
+                () => AccountManager.makeRequest(url, body, headers),
+                AccountManager.shouldRetry,
+                AccountManager.MAX_RETRIES,
+                AccountManager.COMMAND_TIMEOUT,
+                AccountManager.TOTAL_TIMEOUT
+            );
+            console.log(`${AccountManager.LOG_TAG}: Update Reg code successful`);
+        } catch (error) {
+            console.log(
+                `${AccountManager.LOG_TAG}: Update Reg code failed with error=`,
+                error instanceof Error ? error.message : 'UNKNOWN'
+            );
+            throw error;
+        }
+    }
+
+    /**
+     * Sign out user - continues with local sign out even if server fails
+     */
+    public static async signOut(authToken: string): Promise<void> {
+        console.log(`${AccountManager.LOG_TAG}: In sign out with auth token = ${authToken}`);
+        
+        const url = `${AccountManager.BASE_URL}/signout`;
+        console.log(`${AccountManager.LOG_TAG}: fetch url = ${url}`);
+
+        const headers = {
+            'Authorization': authToken,
+            'Content-Type': 'application/json'
+        };
+
+        const body = {
+            deviceId: await getDeviceId()
+        };
+
+        try {
+            await AccountManager.executeWithRetry(
+                () => AccountManager.makeRequest(url, body, headers),
+                AccountManager.shouldRetrySignOut,
+                AccountManager.MAX_RETRIES,
+                AccountManager.COMMAND_TIMEOUT,
+                AccountManager.TOTAL_TIMEOUT
+            );
+            console.log(`${AccountManager.LOG_TAG}: Sign out successful`);
+        } catch (error) {
+            console.log(
+                `${AccountManager.LOG_TAG}: Sign out failed with error=`,
+                error instanceof Error ? error.message : 'UNKNOWN'
+            );
+            // Don't throw error for sign out to allow local sign out to proceed
+        }
+    }
+
+    private static async makeRequest(
```

Comment:
`makeRequest` should also be a utility function

Reply:
DONE